### PR TITLE
Refactor the annotation score for cell lines, genes, and tissues to support multiple selections

### PR DESCRIPTION
**Pull request by oltionchampari**
Coming from Caleydo/tdp_bi_bioinfodb#1143

### Summary 
* Refactored the `cell line | gene | tissue` annotation scores to support multiple selection 

#### before 
---
an_col_before

#### after 
---
ann_col_after

* Changed the button text to `Add`  for all score dialogs (i could not find any mention in any PRs that the button texts were explicitly defined)

#### before
---
adc_before
agc_before
sc_before

#### after
---
button_after

* Added a new i18next namespace. The rest of the strings we can extract in another issue
* Fixed the tours to work with the new dialogs and button names

### CODE REVIEW

File: src/phovea.ts [hunk 1]
```diff
@@ -4,8 +4,10 @@
  * Licensed under the new BSD license, available at http://caleydo.org/license
  **************************************************************************** */
 
+import {EP_PHOVEA_CORE_LOCALE, ILocaleEPDesc, PluginRegistry} from 'phovea_core';
+
 //register all extensions in the registry following the given pattern
-module.exports = function (registry) {
+export default function (registry) {
```
Comment (oltionchampari) [inline src/phovea.ts:10]: This is  **.ts** file we should use ES6 Modules

File: src/scores/AnnotationColumn.ts [hunk 2]
```diff
@@ -33,9 +33,9 @@ export class AnnotationColumn extends ABooleanScore implements IScore<number> {
   protected get columnName() {
     return 'namedset_containment';
   }
-  static createAnnotationColumnScore(data, pluginDesc: IPluginDesc) {
+  static createAnnotationColumnScore(data: IAnnotationColumnParam[], pluginDesc: IPluginDesc) {
     const {primary} = ScoreUtils.selectDataSources(pluginDesc);
-    return new AnnotationColumn(data, primary);
+    return data.map((d) => new AnnotationColumn(d, primary));
```
Comment (thinkh) [inline src/scores/AnnotationColumn.ts:38]: This does not work with existing provenance graphs and causes errors!

grafik

Please check if `data` is an array and handle both cases (single data and data array).

**Review by thinkh — CHANGES_REQUESTED** [review]
Thanks for your changes. In general your changes work for me. However, I found some problems with existing sessions. Please fix the remaining bugs.

Special thanks that you remembered to adapt the tour accordingly. 👍

File: src/scores/GeneSignatureScore.ts [hunk 3]
```diff
@@ -77,37 +77,45 @@ export class GeneSignatureScore implements IScore<number> {
     return RestBaseUtils.getTDPScore(this.dataSource.db, `${this.dataSource.base}_gene_signature_score`, params);
   }
 
-  static createGeneSignatureScore(data: IGeneSignatureData, pluginDesc: IPluginDesc) {
+  static createGeneSignatureScore(data: IGeneSignatureData[], pluginDesc: IPluginDesc) {
     const {primary} = ScoreUtils.selectDataSources(pluginDesc);
-    return new GeneSignatureScore(data.params, primary, data.options);
+    return data.map(({params, options}) => new GeneSignatureScore(params, primary, options));
```
[the remark this answers is not in the document]
Comment (oltionchampari) [reply src/scores/GeneSignatureScore.ts:83]: @thinkh Thanks for thinking of this. Implemented the requested changes.

**Review by thinkh — APPROVED** [review]
Thanks for your fixes. It works now as expected. 👍

**Pull request merged**